indico/modules/rb_new/schemas.py

```diff
@@ -133,7 +136,9 @@ class ReservationDetailsSchema(mm.ModelSchema):
     can_edit = Function(lambda booking: booking.can_edit(session.user))
     can_reject = Function(lambda booking: booking.can_reject(session.user))
     state = EnumField(ReservationState)
-    is_linked_to_event = Function(lambda booking: booking.event_id is not None)
+    is_linked_to_object = Function(lambda booking: booking.link_id is not None)
+    link_type = Function(lambda booking: booking.link.link_type.name if booking.link_id else None)
```

Comment:
Would be much nicer if we could use an `EnumField` here... But I guess marshmallow can't handle this yet due to the nesting? (marshmallow-code/marshmallow#611)

Reply:
you can also use `if booking.link` here; sqlalchemy is smart enough to not trigger a load if the FK is None

Reply:
actually, what about making link a `Nested` here? then you can use a subschema with an EnumField